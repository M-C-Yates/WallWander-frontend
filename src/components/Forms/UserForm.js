import React, { useState } from 'react';
import { emailValidation, passwordValidation } from './UserValidation';

export const UserForm = (props) => {
	const [emailError, setEmailError] = useState('');
	const [passwordError, setPasswordError] = useState('');

	const onEmailChange = (e) => {
		setEmailError(emailValidation(e.target.value));
		if (emailError) return;
		props.setEmail(e.target.value);
	};
	const onPassChange = (e) => {
		setPasswordError(passwordValidation(e.target.value));
		if (passwordError) {
			if (passwordError === 'Please enter a password.') {
				props.setPassword('');
			}
		}
		props.setPassword(e.target.value);
	};
	const buttonText = props.button;

	return (
		<form className="form" onSubmit={props.submit} method="post">
			<div className="form-box">
				<label className="form-label" htmlFor="email">
					Email
				</label>
				<input
					className="form-input"
					id="email"
					name="email"
					onChange={onEmailChange}
					placeholder="hello@email.com"
					required
					type="email"
					value={props.email}
				/>
				{emailError && <p className="form-error">{emailError}</p>}
			</div>

			<div className="form-box">
				<label className="form-label" htmlFor="password">
					Password
				</label>
				<input
					className="form-input"
					id="password"
					minLength="8"
					name="password"
					onChange={onPassChange}
					placeholder="123abc"
					required
					type="password"
					value={props.password}
				/>
				{passwordError && <p className="form-error">{passwordError}</p>}
			</div>
			<button type="submit" className="btn btn-form">
				{buttonText}
			</button>
		</form>
	);
};

export default UserForm;
