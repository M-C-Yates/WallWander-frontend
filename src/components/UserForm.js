import React from 'react';

export const UserForm = (props) => {
	const buttonText = props.button;
	return (
		<form className="form" onSubmit={props.submit} method="post">
			<div className="form-box">
				<label className="form-label" htmlFor="email">
					Email
				</label>
				<input
					className="form-input"
					name="email"
					type="email"
					placeholder="hello@email.com"
					onChange={(e) => props.setEmail(e.target.value)}
				/>
			</div>

			<div className="form-box">
				<label className="form-label" htmlFor="password">
					Password
				</label>
				<input
					className="form-input"
					name="password"
					type="password"
					placeholder="123abc"
					onChange={(e) => props.setPassword(e.target.value)}
					minLength="8"
				/>
			</div>
			<button className="btn btn-form">{buttonText}</button>
		</form>
	);
};
