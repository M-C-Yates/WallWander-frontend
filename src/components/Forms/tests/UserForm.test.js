import React from 'react';
import { render, cleanup } from '@testing-library/react';

import UserForm from '../UserForm';

afterEach(cleanup);

describe('UserForm', () => {
	it('should match snapshot', () => {
		const { asFragment } = render(<UserForm />);
		expect(asFragment).toMatchSnapshot();
	});
});
