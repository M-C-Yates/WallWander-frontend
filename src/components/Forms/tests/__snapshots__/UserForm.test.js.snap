// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm should match snapshot 1`] = `[Function]`;
